src/tx/applyTx/checkSpendCond.js

```diff
@@ -96,67 +112,145 @@ module.exports = async (state, tx, bridgeState, nodeConfig = {}) => {
   addColors(colorMap, await getColors(bridgeState, true), NFT_COLOR_BASE);
+  addColors(colorMap, await getColors(bridgeState, false, true), NST_COLOR_BASE);
 
-  // TODO make inputMap
+  const tokenDataMap = {};
   // convenience mapping from inputs to the previous outputs
   const inputMap = {};
-  for (let i = 0; i < tx.inputs.length; i += 1) {
-    inputMap[tx.inputs[i].prevout.getUtxoId()] =
-      state.unspent[tx.inputs[i].prevout.hex()];
-  }
 
-  // creating a new VM instance
-  const vm = new VM({ hardfork: 'petersburg' });
-  // creating the reactor account with some wei
-  const reactorAccount = new Account();
-  reactorAccount.balance = '0xf00000000000000001';
-  await setAccount(reactorAccount, REACTOR_ADDR, vm.stateManager);
+  for (let i = 0; i < txInputLen; i += 1) {
+    const unspent = state.unspent[tx.inputs[i].prevout.hex()];
+    const tokenId = '0x' + utils.setLengthLeft(
+      utils.toBuffer(`0x${BigInt(unspent.value).toString(16)}`),
+      32
+    ).toString('hex');
+
+    inputMap[tx.inputs[i].prevout.getUtxoId()] = unspent;
+    tokenDataMap[tokenId] = unspent.data;
+  }
 
-  // deploying colors and mint tokens
-  let results;
+  // for deploying colors and mint tokens
   let nonceCounter = 0;
 
-  const sigHashBuf = tx.sigHashBuf();
   const insValues = Object.values(inputMap).reduce(groupValuesByColor, {});
   const outsValues = tx.outputs.reduce(groupValuesByColor, {});
-  // eslint-disable-next-line  guard-for-in
+  const toMint = [];
+
+  // eslint-disable-next-line guard-for-in
   for (const color in insValues) {
-    // eslint-disable-next-line no-await-in-loop
-    await setAccountCode(erc20Code, colorMap[color], vm.stateManager); // eslint-disable-line no-await-in-loop
+    const contractAddr = colorMap[color];
+    const inputValues = insValues[color];
+    const outputValues = outsValues[color];
+    const nst = isNST(color);
+    const nft = isNFT(color);
+
+    let callData;
+    let bytecode;
+
+    if (nst || nft) {
+      for (const valueSet of outputValues) {
+        const tokenId = utils.setLengthLeft(
+          utils.toBuffer(`0x${BigInt(valueSet).toString(16)}`),
+          32
+        );
+
+        if (nst) {
+          const val = '0x' + tokenId.toString('hex');
+          const tokenData = utils.toBuffer(tokenDataMap[val]);
+          callData = Buffer.concat([ERC1948_MINT_FUNCSIG, sigHashBuf, tokenId, tokenData]);
+          bytecode = ERC1948_BYTECODE;
+        }
+        if (nft) {
+          callData = Buffer.concat([ERC721_MINT_FUNCSIG, sigHashBuf, tokenId]);
+          bytecode = ERC721_BYTECODE;
+        }
+
+        toMint.push({ contractAddr, callData, bytecode, color });
+      }
+    } else {
+      // default to ERC20
+      // it does not need an output actually,
+      // input can be burned completely
+      const value = utils.setLengthLeft(
+        utils.toBuffer(`0x${BigInt(outputValues || inputValues).toString(16)}`),
```

Comment:
possible text for comment
usually we burn the output, so that `token.balance(address(this))` returns the balance after gas subtracted. 
this is a special case where there is a color input, but no color output. in that case we mint it, and `token.balance(address(this))` will return it.